includes/class-wc-gateway-swedbank-pay-checkout.php

```diff
@@ -0,0 +1,1837 @@
+	/**
+	 * If There are no payment fields show the description if set.
+	 */
+	public function payment_fields() {
+
+	}
+
+	/**
+	 * Validate Frontend Fields
+	 * @return bool|void
+	 */
+	public function validate_fields() {
+		//
+	}
+
+	/**
+	 * Thank you page
+	 *
+	 * @param $order_id
+	 */
+	public function thankyou_page( $order_id ) {
+		WC()->session->__unset( 'payex_paymentorder_id' );
+	}
+
+	/**
+	 * Process Payment
+	 *
+	 * @param int $order_id
+	 *
+	 * @return array|false
+	 */
+	public function process_payment( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		// Order Info
+		$info = $this->get_order_info( $order );
+
+		// Get Order Items
+		$items = apply_filters( 'swedbank_pay_checkout_order_items', [], $order );
+
+		if ( isset( $_POST['is_update'] ) ) {
+			$order->calculate_totals( true );
+
+			// Get Payment Order ID
+			$payment_id = get_post_meta( $order_id, '_payex_paymentorder_id', true );
+			if ( empty( $payment_id ) ) {
+				$payment_id = WC()->session->get( 'payex_paymentorder_id' );
+			}
+
+			if ( empty( $payment_id ) ) {
+				// PaymentOrder is unknown
+				return false;
+			}
+
+			$result     = $this->request( 'GET', $payment_id );
+			$js_url     = self::get_operation( $result['operations'], 'view-paymentorder' );
+			$update_url = self::get_operation( $result['operations'], 'update-paymentorder-updateorder' );
+			if ( empty( $update_url ) ) {
+				throw new Exception( 'Order update is not available.' );
+			}
+
+			// Don't update if amount is not changed
+			if ( (int) $result['paymentOrder']['amount'] === round( $order->get_total() * 100 ) ) {
+				return [
+					'result'               => 'success',
+					'redirect'             => '#!swedbank-pay-checkout',
+					'is_swedbank_pay_checkout' => true,
+					'js_url'               => $js_url,
+					'payment_id'           => $result['paymentOrder']['id'],
+				];
+			}
+
+			// Update Order
+			$params = [
+				'paymentorder' => [
+					'operation' => 'UpdateOrder',
+					'amount'    => apply_filters( 'swedbank_pay_checkout_order_amount', 0, $order ),
+					'vatAmount' => apply_filters( 'swedbank_pay_checkout_order_vat', 0, $order ),
+					'orderItems' => $items
+				]
+			];
+
+			$result = $this->request( 'PATCH', $update_url, $params );
+
+			// Get JS Url
+			$js_url = self::get_operation( $result['operations'], 'view-paymentorder' );
+
+			return [
+				'result'               => 'success',
+				'redirect'             => '#!swedbank-pay-checkout',
+				'is_swedbank_pay_checkout' => true,
+				'js_url'               => $js_url,
+				'payment_id'           => $result['paymentOrder']['id'],
+			];
+		}
+
+		// Mode that workaround "Order update is not available"
+		if ( isset( $_POST['is_update_backward_compat'] ) ) {
+			$order->calculate_totals( true );
+
+			// Delete Payment Order ID
+			delete_post_meta( $order_id, '_payex_paymentorder_id' );
+		}
+
+		// Check terms_url value
+		if ( parse_url( $this->terms_url, PHP_URL_SCHEME ) !== 'https' ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function parse_url() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)